phpunit/class-rest-block-renderer-controller-test.php

```diff
@@ -0,0 +1,297 @@
+		wp_set_current_user( self::$user_id );
+
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( self::$block_name );
+		$defaults   = array();
+		foreach ( $block_type->attributes as $key => $attribute ) {
+			$defaults[ $key ] = isset( $attribute['default'] ) ? $attribute['default'] : null;
+		}
+
+		$request = new WP_REST_Request( 'GET', '/gutenberg/v1/block-renderer/' . self::$block_name );
+		$request->set_param( 'attributes', array() );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 200, $response->get_status() );
+		$data = $response->get_data();
+
+		$this->assertEquals( $defaults, json_decode( $data['rendered'], true ) );
+		$this->assertEquals(
+			json_decode( $block_type->render( $defaults ) ),
+			json_decode( $data['rendered'] )
+		);
+	}
+
+	/**
+	 * Check getting item with attributes provided.
+	 *
+	 * @covers WP_REST_Block_Renderer_Controller::get_item()
+	 */
+	public function test_get_item() {
+		wp_set_current_user( self::$user_id );
+
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( self::$block_name );
+		$attributes = array(
+			'some_int'    => '123',
+			'some_string' => 'foo',
+			'some_array'  => array( 1, '2', 3 ),
+		);
+
+		$expected_attributes               = $attributes;
+		$expected_attributes['some_int']   = (int) $expected_attributes['some_int'];
+		$expected_attributes['some_array'] = array_map( 'intval', $expected_attributes['some_array'] );
+
+		$request = new WP_REST_Request( 'GET', '/gutenberg/v1/block-renderer/' . self::$block_name );
+		$request->set_param( 'attributes', $attributes );
+		$response = $this->server->dispatch( $request );
+		$this->assertEquals( 200, $response->get_status() );
+		$data = $response->get_data();
+
+		$this->assertEquals( $expected_attributes, json_decode( $data['rendered'], true ) );
+		$this->assertEquals(
+			json_decode( $block_type->render( $attributes ), true ),
+			json_decode( $data['rendered'], true )
+		);
+	}
+
+	/**
+	 * Get item schema.
+	 *
+	 * @covers WP_REST_Block_Renderer_Controller::get_item_schema()
+	 */
+	public function test_get_item_schema() {
+		$request  = new WP_REST_Request( 'OPTIONS', '/gutenberg/v1/block-renderer/' . self::$block_name );
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEqualSets( array( 'GET' ), $data['endpoints'][0]['methods'] );
+		$this->assertEqualSets(
+			array( 'name', 'context', 'attributes' ),
+			array_keys( $data['endpoints'][0]['args'] )
+		);
+		$this->assertEquals( 'object', $data['endpoints'][0]['args']['attributes']['type'] );
+
+		$this->assertArrayHasKey( 'schema', $data );
+		$this->assertEquals( 'rendered-block', $data['schema']['title'] );
+		$this->assertEquals( 'object', $data['schema']['type'] );
+		$this->arrayHasKey( 'rendered', $data['schema']['properties'] );
+		$this->arrayHasKey( 'string', $data['schema']['properties']['rendered']['type'] );
+	}
+
+	/**
+	 * NA.
+	 */
```

Comment:
I'd drop the placeholder comments, since they can throw our linter off and later slip through review.